### PR TITLE
SSH refurbish: write hostnames correctly

### PR DESCRIPTION
**Pull request by johncmerfeld**
## Description of the change

Currently when we provide the ssh daemon with a list of hostnames to write to known_hosts, we use the `parsedTargetString`, which as we all know cannot handle targets with periods in their name. Instead, we should let the backend parse the target string for us and use the returned `targetName` as our hostname.

Also fixes an issue where errors in the ssh proxy handler were crashing the zli instead of exiting cleanly

Full family of PRs:
  - https://github.com/bastionzero/webshell-backend/pull/1425
  - https://github.com/bastionzero/bzero/pull/179
  - https://github.com/bastionzero/zli/pull/576
  - https://github.com/bastionzero/webshell-common-ts/pull/96

## Testing

Describe how to test this PR....

**backend branch:** 
**bzero branch:** 
**charts branch:** 

#### Ready to run system tests?

- [x] Yes

## Relevant release note information

Release Notes: write SSH hostnames correctly

## Related JIRA tickets

Relates to JIRA: CWC-2204, CWC-2206

## Relevant HotFix if applicable

Relevant HotFix PR Number:

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [x] No

If yes, please explain:

### CODE REVIEW

File: src/utils/unit-test-utils.ts [hunk 1]
```diff
@@ -346,6 +346,7 @@ export const mockUniversalConnectionRequest: CreateUniversalConnectionRequest =
 export const mockUniversalConnectionResponse: CreateUniversalConnectionResponse = {
     connectionId: 'f11ae2c2-48a7-4a29-b306-8bc83da3c045',
     targetId: '9f728982-3691-4e71-9533-1892d5c65eb9',
+    targetName: 'test-target',
```
Comment (SaiDadireddy) [inline src/utils/unit-test-utils.ts:349]: Can you also update the targetName in the `mockUniversalConnectionRequest` above please? Both of these mocks are used in the [connect unit test](https://github.com/bastionzero/zli/blob/e3905895c2ecd2b0e1526c053f96f245fb6929ce/src/handlers/generate/generate-ssh-config.test.ts#L164-L201), so better to keep them both consistent. Although, won't affect anything if you don't do your choice :)

Comment (johncmerfeld) [reply src/utils/unit-test-utils.ts:349]: Good call, will do!

Comment (johncmerfeld) [reply src/utils/unit-test-utils.ts]: ec7460f

**Review by SaiDadireddy — APPROVED** [review]
lgtm no blockers

**Pull request merged**